src/parser/jobs/brd/modules/Snapshots.tsx

```diff
@@ -0,0 +1,260 @@
+import React from 'react'
+import {Trans, Plural} from '@lingui/react'
+import {t} from '@lingui/macro'
+import {Table, Accordion} from 'semantic-ui-react'
+
+import Module, {dependency} from 'parser/core/Module'
+import {CastEvent} from 'fflogs'
+import STATUSES, {Status} from 'data/STATUSES'
+import {SNAPSHOT_BLACKLIST} from 'parser/jobs/brd/modules/SnapshotBlacklist'
+import ACTIONS from 'data/ACTIONS'
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import NormalisedMessage from 'components/ui/NormalisedMessage'
+import Combatants from 'parser/core/modules/Combatants'
+import {AbilityEvent} from 'fflogs'
+import Util from './Util'
+import {Data} from 'parser/core/modules/Data'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+
+import styles from './Snapshots.module.css'
+
+const SNAPSHOTTERS = [
+	ACTIONS.IRON_JAWS.id,
+	ACTIONS.CAUSTIC_BITE.id,
+	ACTIONS.STORMBITE.id,
+]
+
+const DOT_STATUSES = [
+	STATUSES.CAUSTIC_BITE.id,
+	STATUSES.STORMBITE.id,
+]
+
+const PERSONAL_STATUSES = [
+	STATUSES.MEDICATED.id,
+	STATUSES.RAGING_STRIKES.id,
+]
+
+const BAD_REFRESH_SEVERITY = {
+	1: SEVERITY.MINOR,
+	2: SEVERITY.MEDIUM,
+	3: SEVERITY.MAJOR,
+}
+
+interface Snapshot {
+	// The event that applied the DoT(s).
+	snapEvent: CastEvent
+
+	// All statuses observed at the time of casting the snapshot.
+	statuses: Status[]
+
+	// Indicates whether the snap was an IJ that did not refresh DoTs.
+	badRefresh: boolean
+}
+
+class Target {
+	public name: string
+	public key: string
+	public snapshots: Snapshot[] = []
+	private currentSnapshot?: Snapshot
+	private statuses = new Set<Status>()
+
+	constructor(name: string, key: string) {
+		this.name = name
+		this.key = key
+	}
+
+	public addStatus(status: Status) {
+		this.statuses.add(status)
+	}
+
+	public removeStatus(status: Status) {
+		this.statuses.delete(status)
+	}
+
+	public addSnapshot(snap: CastEvent, playerStatuses: Status[]) {
+		// If IJ didn't refresh a caustic bite, it was a misuse
+		// (IJ on storm bite is technically potency neutral, though odd)
```

Comment:
Do we care about potency neutrality, or encouraging players to best practice? I don't think it matters much, but worth a thought.